src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -43,14 +43,21 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
         public async Task<ExecutionResult> InitializeAsync() {
             try {
                 if (!Session.IsHostRunning) {
-                    await Session.StartHostAsync(new RHostStartupInfo {
+                    var startupInfo = new RHostStartupInfo {
                         Name = "REPL",
                         RBasePath = _settings.RBasePath,
                         RHostCommandLineArguments = _settings.RCommandLineArguments,
                         CranMirrorName = _settings.CranMirror,
                         CodePage = _settings.RCodePage,
-                        WorkingDirectory = _settings.WorkingDirectory
-                    }, new RSessionCallback(CurrentWindow, Session, _settings, _coreShell));
+                        WorkingDirectory = _settings.WorkingDirectory,
+                    };
+
+                    IInteractiveWindowVisualComponent visualComponent;
```

Comment:
Sure, if we don't care about a bit of UI specific code there, that will simplify things quite a bit.